### PR TITLE
Add folders to myFolders page

### PR DESCRIPTION
**Pull request by Froggerson**
image
Screenshot 2020-08-07 at 8 10 05 AM
So far a way to integrate the userid has not been completed yet.

### CODE REVIEW

File: frontend/src/main-components/MyFoldersContent.js [hunk 1]
```diff
@@ -9,11 +10,54 @@ const MyFoldersContent = (props) => {
...
+
+  //Fetch data from the UserFolders servlets to get the user folders
+  //Currently there is no way to get the userid of the user
+  try {
+    fetch(`/userfolder?userKey=${userKey}`, { method: "GET" })
+      .then((result) => result.json())
+      .then((data) => {
+        console.log(data);
+        folderData = data;
+        const folders = folderData.map((folder) => (
```
Comment (Froggerson) [inline frontend/src/main-components/MyFoldersContent.js]: The folderData isn't really needed here.

File: frontend/src/main-components/MyFoldersContent.js [hunk 2]
```diff
@@ -9,11 +10,54 @@ const MyFoldersContent = (props) => {
...
+          <Folder
+            folderKey={folder.folderKey}
+            name={folder.folderName}
+            language={folder.folderDefaultLanguage}
+            key={folder.folderKey}
+          />
```
Comment (ethyus) [inline frontend/src/main-components/MyFoldersContent.js]: Shouldn't there only be one "folder.folderKey"?

File: frontend/src/main-components/MyFoldersContent.js [hunk 3]
```diff
@@ -9,11 +10,54 @@ const MyFoldersContent = (props) => {
...
+    display: "flex",
+    flexFlow: "row wrap",
+    justifyContent: "space-around",
+    alignItems: "start",
+  };
+
+  let folders;
+
+  //Fetch data from the UserFolders servlets to get the user folders
+  //Currently there is no way to get the userid of the user
```
Comment (ethyus) [inline frontend/src/main-components/MyFoldersContent.js]: What do you mean by there is no way to get the user id? Do you mean the userKey?

File: frontend/src/flashcards/Folder.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "center",
+  alignItems: "center",
+};
+const outerStyle = {
+  width: "300px",
+  height: "350px",
+  margin: "20px",
+};
```
Comment (ESaracay) [inline frontend/src/flashcards/Folder.js]: If the CSS got any longer than this then I would consider maybe creating a seperate flashcards.css file.

File: frontend/src/main-components/MyFoldersContent.js [hunk 5]
```diff
@@ -9,11 +10,54 @@ const MyFoldersContent = (props) => {
...
+  }
+
+  // The page should show different text depending on whether or not folders are available
+  let headingText;
+  if (folders) {
+    headingText = `You have ${folders.length} Folders`;
+  } else {
+    headingText =
+      "You have no folders at the moment. Please make a new folder by clicking on the side menu.";
+  }
```
Comment (ESaracay) [inline frontend/src/main-components/MyFoldersContent.js]: You can shorten this up by saying let headingText= "You have no folders ...",  which lets you get rid of the else statement.

File: frontend/src/main-components/MyFoldersContent.js (hunk 2 above)
Comment (Froggerson) [reply frontend/src/main-components/MyFoldersContent.js]: I ran into some compile errors while passing in key as a prop, so I added the folderKey prop as way to circumvent that. I can change the prop name!

File: frontend/src/main-components/MyFoldersContent.js (hunk 3 above)
Comment (Froggerson) [reply frontend/src/main-components/MyFoldersContent.js]: Yeah.

File: frontend/src/flashcards/Folder.js (hunk 4 above)
Comment (Froggerson) [reply frontend/src/flashcards/Folder.js]: alright!

File: frontend/src/main-components/MyFoldersContent.js (hunk 5 above)
Comment (Froggerson) [reply frontend/src/main-components/MyFoldersContent.js]: ok!

**Review by ESaracay — APPROVED** [review]

**Comment by tylerhou** [comment]
This PR seems to duplicate code found in #21. Can you close one PR/change the diffbase?

**Pull request merged**